pkg/stash/with_redis.go

```diff
@@ -14,11 +14,12 @@ import (
 
 // WithRedisLock returns a distributed singleflight
 // using an redis cluster. If it cannot connect, it will return an error.
-func WithRedisLock(endpoint string, checker storage.Checker) (Wrapper, error) {
+func WithRedisLock(endpoint string, password string, checker storage.Checker) (Wrapper, error) {
 	const op errors.Op = "stash.WithRedisLock"
 	client := redis.NewClient(&redis.Options{
-		Network: "tcp",
-		Addr:    endpoint,
+		Network:  "tcp",
+		Addr:     endpoint,
+		Password: password,
```

Comment:
@mplachter if there is no password set in the redis server config, will passing `""` here be ok?

Sidenote: I'm not sure if we test this case in `with_redis_test.go` below, but we can add that in a follow-up

Reply:
Hey @arschles,

No problem happy to help :)

So i'm pretty sure this is being testing in `with_redis_test.go` already as there is no env variable set for `ATHENS_REDIS_PASSWORD`, this should cause os.Getenv to return an empty string which is passed into WithRedisLock.

What still needs to be tested is the ability to test with `password` working correctly,  the docker `redis` image does not have a straight forward interface for setting this password, instead they support copying a redis [conf](http://download.redis.io/redis-stable/redis.conf) over to the container.

```
FROM redis
COPY redis.conf /usr/local/etc/redis/redis.conf
CMD [ "redis-server", "/usr/local/etc/redis/redis.conf" ]
```

I do not mind doing this work just would like to know where you prefer the config to be stored.

Reply:
@mplachter I missed that about `with_redis_test.go` - thanks for setting me straight 😄 

Thanks for volunteering to expand the tests for password, too. You can put the `redis.conf` file in the `test/` directory. And if you'd like to get this merged, it's fine to do it in a follow-up PR.

Just let me know what you'd like to do

Reply:
Hey @arschles lets add it in the same PR.